File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_index = {}
        for i in range(0, len(nums)):
            complement = target - nums[i]
            if complement in num_index:
                return [num_index[complement], i]
            num_index[nums[i]] = i
        
        return None
                

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
	numIndex := make(map[int]int)
	for index, value := range nums {
		complement := target - value
		mapValue, ok := numIndex[complement]
		if ok {
			return []int{mapValue, index}
		}
		numIndex[value] = index
	}

	return nil
}